addons/lunch/report/lunch_product_report.py

```diff
@@ -55,13 +56,17 @@ def _search_is_available_at(self, operator, value):
     def write(self, values):
         user_id = self.env.user.id
         if 'is_favorite' in values:
-            for report in self:
-                if values['is_favorite']:
-                    self.env.cr.execute('''INSERT INTO lunch_product_favorite_user_rel(product_id, user_id) VALUES (%s, %s)''',
-                                (report.product_id, user_id))
-                else:
-                    self.env.cr.execute('''DELETE FROM lunch_product_favorite_user_rel WHERE product_id=%s AND user_id=%s''',
-                                (report.product_id, user_id))
+            # Use sudo here, because we cannot allow everyone to modify lunch.product
+            # but everyone needs to be able to favorite a product
+            products = self.env['lunch.product'].browse(self.mapped('product_id')).sudo()
```

Comment:
does the inverse M2M  (from users to product) exists? Isn't it better to write on this instead of the sudo?

Reply:
There is no inverse field currently declared on the user, but yes I agree that it would be better than using sudo

Reply:
Maybe add it in master before freeze...

Reply:
I have a branch (https://github.com/odoo-dev/odoo/tree/master-imp-lunch-rli) in master that fixes that and the FIXME comment